hw/drivers/sensors/lsm6dsl/include/lsm6dsl/lsm6dsl.h

```diff
@@ -0,0 +1,795 @@
+ *
+ * @param itf The sensor interface
+ * @param src Ptr to return 4 interrupt sources in
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_clear_int(struct lsm6dsl *lsm6dsl, struct int_src_regs *int_src);
+
+/**
+ * Set interrupt pin configuration for interrupt
+ *
+ * @param lsm6dsl The device
+ * @param interrupt pin
+ * @param interrupt mask
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_set_int_pin_cfg(struct lsm6dsl *lsm6dsl, uint8_t int_pin,
+                            uint8_t int_mask);
+
+/**
+ * Set orientation configuration
+ *
+ * @param lsm6dsl The device
+ * @param cfg orientation settings to set
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_set_orientation(struct lsm6dsl *lsm6dsl,
+                            const struct lsm6dsl_orientation_settings *cfg);
+
+/**
+ * Get orientation configuration
+ *
+ * @param lsm6dsl The device
+ * @param cfg orientation settings to fill
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_get_orientation_cfg(struct lsm6dsl *lsm6dsl,
+                                struct lsm6dsl_orientation_settings *cfg);
+
+/**
+ * Set tap detection configuration
+ *
+ * @param lsm6dsl The device
+ * @param cfg new tap settings
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_set_tap_cfg(struct lsm6dsl *lsm6dsl,
+                        const struct lsm6dsl_tap_settings *cfg);
+
+/**
+ * Get tap detection config
+ *
+ * @param lsm6dsl The device
+ * @param ptr to the tap settings
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_get_tap_cfg(struct lsm6dsl *lsm6dsl,
+                        struct lsm6dsl_tap_settings *cfg);
+
+/**
+ * Set free-fall detection configuration
+ *
+ * @param lsm6dsl The device
+ * @param ff free fall settings
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_set_free_fall(struct lsm6dsl *lsm6dsl, struct lsm6dsl_ff_settings *ff);
+
+/**
+ * Get free-fall detection config
+ *
+ * @param lsm6dsl The device
+ * @param ff ptr to free-fall configuration
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_get_free_fall(struct lsm6dsl *lsm6dsl, struct lsm6dsl_ff_settings *ff);
+
+/**
+ * Set Wake Up Duration
+ *
+ * @param lsm6dsl The device
+ * @param wk configuration to set
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_set_wake_up(struct lsm6dsl *lsm6dsl, struct lsm6dsl_wk_settings *wk);
+
+/**
+ * Get Wake Up Duration
+ *
+ * @param lsm6dsl The device
+ * @param ptr to wake up configuration
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int lsm6dsl_get_wake_up(struct lsm6dsl *lsm6dsl, struct lsm6dsl_wk_settings *wk);
+
+/**
+ * Enable sensor fifo interrupt
+ *
+ * @param sensor device
+ * @param lsm6dsl_cfg configuration
+ *
+ * @return 0 on success, non-zero on failure
+ */
+int enable_fifo_interrupt(struct sensor *sensor, sensor_type_t type,
```

Comment:
lsm6dsl_enable_fifo_interrupt ?  (or just make it static since it doesn't seem to be used outside of c file)